### PR TITLE
Associations setup

### PR DESCRIPTION
**Pull request by afizsavage**
- Create classes with the following names: Classroom, Books, and Rental
- Create the has-many/belongs-to relationship between Classroom and Student classes
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (Emmyn5600) [inline rental.rb]: - [ ] Book and person has to be the attr_accessor, not attr_reader.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  def add_book(book)
+    @book = book
+    book.rentals.push(self) unless book.rentals.include?(self)
+  end
+
+  def add_person(person)
+    @person = person
+    person.rentals.push(self) unless person.rentals.include?(self)
+  end
```
Comment (Emmyn5600) [inline rental.rb:22]: - [ ] The above implementation was correct to add or push both book and person.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (Emmyn5600) [inline book.rb]: - [ ] rentals have to be attr_accessor too.

- [ ] From the def add_rental implementation, you have to implement it in the below one.

```ruby
  def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @afizsavage 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:
- The database is well structured/created. :heavy_check_mark:

## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
   def initialize(age, name = 'unknown', parent_permission: true)
     @id = Random.rand(1...1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
 
+  def add_rental(_rental)
```
Comment (mmsesay) [inline person.rb]: - [ ] I got an error when I tried to add a rental for a person. This method should take to arguments, `book` & `date`. As of now, your method only takes one argument. And when I pass the one argument, it also throw an error.

See screenshots:

With two arguments passed|
:--------------------------------------:|
a1

With one argument passed|
:-----------------------------------:|
a1

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @afizsavage, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: 
Approved



Hi @afizsavage , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**